modules/commons/src/main/java/org/apache/synapse/commons/json/JsonUtil.java

```diff
@@ -586,6 +604,31 @@ public static OMElement getNewJsonPayload(MessageContext messageContext, InputSt
                 jsonElement = JSON_OBJECT;
                 messageContext.setProperty(ORG_APACHE_SYNAPSE_COMMONS_JSON_IS_JSON_OBJECT, true);
             }
+            if (isValue) {
+                String jsonString = "";
+                try {
+                    jsonString = IOUtils.toString(json, "UTF-8");
+
+                    messageContext.setProperty(ORG_APACHE_SYNAPSE_COMMONS_JSON_IS_JSON_OBJECT, false);
+                    if (jsonString.startsWith("\"") || jsonString.equals("true") || jsonString.equals("false") ||
+                            jsonString.equals("null") || jsonString.matches("-?\\d+(\\.\\d+)?")) {
+                        jsonElement = JSON_VALUE;
+                    } else {
+                        logger.error(
+                                "#getNewJsonPayload. Could not save JSON payload. Invalid input stream found. " +
+                                        "MessageID: " +
+                                        messageContext.getMessageID());
+                        throw new AxisFault("Payload is not a JSON string.");
```

Comment:
Why are we throwing AxisFault here? Can't we throw something appropriate, or perhaps SynapseException?